File: laravel-kostFinder/app/Models/Review.php
<?php

namespace App\Models;

use MongoDB\Laravel\Eloquent\Model;

class Review extends Model
{
    protected $connection = 'mongodb';
    protected $collection = 'reviews';

    protected $fillable = [
        'user_id',
        'kost_id',
        'rating',
        'komentar',
        'status',   // Menunggu | Disetujui | Ditolak
    ];

    protected $casts = [
        'rating' => 'integer',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function kost()
    {
        return $this->belongsTo(Kost::class, 'kost_id', '_id');
    }
}

File: flutter_kostfinder/lib/models/review.dart
class Review {
  final String id;
  final dynamic userId;
  final String kostId;
  final int rating;
  final String komentar;

  Review({
    required this.id,
    required this.userId,
    required this.kostId,
    required this.rating,
    required this.komentar,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id']?.toString() ?? '',
      userId: json['user_id'],
      kostId: json['kost_id']?.toString() ?? '',
      rating: (json['rating'] ?? 0).toInt(),
      komentar: json['komentar'] ?? '',
    );
  }
}
